src/entity/Company.java

```diff
@@ -1,35 +1,36 @@
 package entity;
 
-import java.util.ArrayList;
+import parser.CarsFileReader;
+import parser.CarsParser;
+
 import java.util.List;
-import java.util.logging.Level;
-import java.util.logging.Logger;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 public class Company {
     private static Company company;
-    private static AtomicBoolean instanceCreated = new AtomicBoolean(false);
+    private static AtomicBoolean instanceCreated = new AtomicBoolean();
```

Comment:
What is the value by default? I think you should return **false** as an argument of the constructor.